packages/abilities/src/store/selectors.ts

```diff
@@ -0,0 +1,82 @@
+/**
+ * WordPress dependencies
+ */
+import { createSelector } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import type {
+	Ability,
+	AbilityCategory,
+	AbilitiesQueryArgs,
+	AbilitiesState,
+} from '../types';
+
+/**
+ * Returns all registered abilities.
+ * Optionally filters by query arguments.
+ *
+ * @param state Store state.
+ * @param args  Optional query arguments to filter. Defaults to empty object.
+ * @return Array of abilities.
+ */
+export const getAbilities = createSelector(
+	(
+		state: AbilitiesState,
+		{ category }: AbilitiesQueryArgs = {}
+	): Ability[] => {
+		const abilities = Object.values( state.abilitiesByName );
+		if ( category ) {
+			return abilities.filter(
+				( ability ) => ability.category === category
+			);
+		}
+		return abilities;
+	},
+	( state: AbilitiesState, category?: string ) => [
+		state.abilitiesByName,
+		category,
+	]
+);
+
+/**
+ * Returns a specific ability by name.
+ *
+ * @param state Store state.
+ * @param name  Ability name.
+ * @return Ability object or null if not found.
+ */
+export function getAbility(
+	state: AbilitiesState,
+	name: string
+): Ability | null {
+	return state.abilitiesByName[ name ] || null;
```

Comment:
Is there a good reason why we're using `null` instead of `undefined` to signify that an ability is not found?

Reply:
We can switch this to `undefined`. 👍